mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMailboxMapper.java

```diff
@@ -200,10 +201,11 @@ private CassandraId retrieveId(SimpleMailbox cassandraMailbox) {
     }
 
     @Override
-    public boolean hasChildren(Mailbox mailbox, char delimiter) {
+    public boolean hasChildren(Mailbox mailbox, PathDelimiter delimiter) {
         return mailboxPathDAO.listUserMailboxes(mailbox.getNamespace(), mailbox.getUser())
             .thenApply(stream -> stream
-                .anyMatch(idAndPath -> idAndPath.getMailboxPath().getName().startsWith(mailbox.getName() + String.valueOf(delimiter))))
+                .anyMatch(idAndPath -> idAndPath.getMailboxPath().getName()
+                    .startsWith(delimiter.appendDelimiter(mailbox.getName()))))
```

Comment:
looks like we just want to know if the string represents several Path parts, right ? What about having that method in PathDelimiter ?

Reply:
No, we want to enforce that these mailboxName are children.